docs/tutorials/sparse/train_gluon.md

```diff
@@ -0,0 +1,469 @@
+
+## Gluon Models for Sparse Data
+
+Our next step is to define a network. We have an input of 1,000,000 features and we want to make a binary prediction. We don't have any spatial or temporal relationships between features, so we'll use a 3 layer fully-connected network where the last layer has 1 output unit (with sigmoid activation). Since we're working with sparse data, we'd ideally like to use network operators that can exploit this sparsity for improved performance and memory efficiency.
+
+Gluon's [`nn.Dense`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/nn.html?highlight=dense#mxnet.gluon.nn.Dense) block can used with [`CSRNDArray`](https://mxnet.incubator.apache.org/api/python/ndarray/sparse.html?highlight=csrndarray#mxnet.ndarray.sparse.CSRNDArray) input arrays but it doesn't exploit the sparsity. Under the hood, [`Dense`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/nn.html?highlight=dense#mxnet.gluon.nn.Dense) uses the [`FullyConnected`](https://mxnet.incubator.apache.org/versions/master/api/python/ndarray/ndarray.html?highlight=fullyconnected#mxnet.ndarray.FullyConnected) operator which isn't optimized for [`CSRNDArray`](https://mxnet.incubator.apache.org/api/python/ndarray/sparse.html?highlight=csrndarray#mxnet.ndarray.sparse.CSRNDArray) arrays. We'll implement a `Block` that does exploit this sparsity, *but first*, let's just remind ourselves of the [`Dense`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/nn.html?highlight=dense#mxnet.gluon.nn.Dense) implementation by creating an equivalent `Block` called `FullyConnected`.
+
+
+```python
+class FullyConnected(mx.gluon.HybridBlock):
+    def __init__(self, in_units, units):
+        super(FullyConnected, self).__init__()
+        with self.name_scope():
+            self._units = units
+            self.weight = self.params.get('weight', shape=(units, in_units),
+                                          init=None, allow_deferred_init=True,
+                                          dtype='float32', stype='default', grad_stype='default')
+            self.bias = self.params.get('bias', shape=(units),
+                                          init='zeros', allow_deferred_init=True,
+                                          dtype='float32', stype='default', grad_stype='default')
+
+    def hybrid_forward(self, F, x, weight, bias):
+        return F.FullyConnected(x, weight, bias, num_hidden=self._units)
+```
+
+Our `weight` and `bias` parameters are dense (see `stype='default'`) and so are their gradients (see `grad_stype='default'`). Our `weight` parameter has shape `(units, in_units)` because the [`FullyConnected`](https://mxnet.incubator.apache.org/versions/master/api/python/ndarray/ndarray.html?highlight=fullyconnected#mxnet.ndarray.FullyConnected) operator performs the following calculation:
+
+$$Y = XW^T + b$$
```

Comment:
Is this supposed to render differently on github or just when rendered on the website?

Reply:
Yep, Github rendering screws up.